extension/content/components/recipes/form/RecipeFormHeader.js

```diff
@@ -71,6 +81,45 @@ export default function RecipeFormHeader() {
           Save
         </IconButton>
       </div>
+      <SaveModal
+        show={showCommentModal}
+        onClose={closeSaveModal}
+        onSave={saveRecipe}
+      />
     </div>
   );
 }
+
+function SaveModal(props) {
+  const { show, onClose, onSave } = props;
+
+  const handleSaveClick = () => {
+    onSave();
+    onClose();
+  };
+
+  return (
+    <Modal show={show} onHide={onClose}>
+      <Modal.Header>
+        <Modal.Title>Save Recipe</Modal.Title>
+      </Modal.Header>
+      <Modal.Body>
+        <GenericField label="Comment on Revision" name="comment" />
```

Comment:
A couple things about this field.

Using a generic field means its going to populate with the previous value which is not the behavior we want. This comment should be blank whenever the user is creating a new revision since it is like a commit message and unique to the revision.

Also I think it should be a textarea not a text input. 

Finally I think we should not have a label and instead have some placeholder text instead:
```js
<textarea placeholder="Please describe the changes you are making&hellip;" />
```